spec/helpers/registration_helper_spec.rb

```diff
@@ -1,7 +1,38 @@
 require 'spec_helper'
 
 describe RegistrationHelper do
-  it 'tests nothing' do
-    skip
+  describe '#apo_list' do
+    let(:perm_keys) { ['sunetid:user', 'workgroup:dlss:mock-group1', 'workgroup:dlss:mock-group2'] }
+
+    it 'runs the appropriate query for the given permission keys' do
+      q = perm_keys.map { |key| %(apo_register_permissions_ssim:"#{key}") }.join(' OR ')
+
+      expect(Dor::SearchService).to receive(:query).with(
+        q,
+        defType: 'lucene',
+        rows: 99999,
+        fl: 'id,tag_ssim,dc_title_tesim',
+        fq: ['objectType_ssim:adminPolicy', '!tag_ssim:"Project : Hydrus"']
+      ).and_return(double(docs: []))
+
+      apo_list(perm_keys)
+    end
+
+    it 'sorts the results and formats them correctly' do
+      result_rows = [
+        {'id' => 1, 'tag_ssim' => 'prefix : suffix', 'dc_title_tesim' => 'z'},
+        {'id' => 2, 'tag_ssim' => 'AdminPolicy : default', 'dc_title_tesim' => 'y'},
+        {'id' => 3, 'tag_ssim' => 'prefix : suffix2', 'dc_title_tesim' => 'x'}
+      ]
+      expect(Dor::SearchService).to receive(:query).and_return(double(docs: result_rows))
+
+      apos = apo_list(perm_keys)
+      expect(apos).to eq [['y', '2'], ['x', '3'], ['z', '1']]
```

Comment:
i prefer it as-is, but would be willing to change this if others would rather follow hound's suggestion.  but i find this much more readable than the suggestion for this particular situation.